Supplementary Methods for:

**Analyses of variant data across the whole genome of the American Shad, _Alosa sapidissima_**

Authors: order? 

**Raw Data**

All project, sample information, and raw sequencing data are publicly available and can be accessed by the following NCBI accession numbers:

| SRA Accession | Data Type       |
| ------------- | -------------   |
|  PRJNA728145  | Reference genome|
| SRX4806914  | Short Insert Library  |
| SRX4806913  | Mate Pair Library  |
| SRX4806912  | Mate Pair Library  |

GNU license? #should we put something like this in?

**Table of Contents**

Data Accession and Cleaning: https://github.com/ea227/A.-sapidissima-Scripts/blob/main/Data%20Accession%20and%20Cleaning.md
Quality Control: https://github.com/ea227/A.-sapidissima-Scripts/blob/main/Quality%20Control.md
Alignment and Processing: https://github.com/ea227/A.-sapidissima-Scripts/blob/main/Alignment%20and%20Processing.md

**Software and Programs Used**

| Program | Version | Citation|
| --------  | ------ | --------|
|SRA Toolkit | 3.1.1 | Edwards, Robert A. (2021). SearchSRA Toolkit. Tools for processing data generated by the Search SRA. version 0.1.0 doi:10.5281/zenodo.5044723|
| NxTrim | 0.4.3 | O’Connell, J., Schulz-Trieglaﬀ, O., Carlson, E., Hims, M.M., Gormley, N.A., Cox, A.J. (2015). NxTrim: optimized trimming of Illumina mate pair reads, Bioinformatics, Volume 31, Issue 12, Pages 2035–2037, https://doi.org/10.1093/bioinformatics/btv057 |
| fastp | 0.20.0 | Chen, Shifu. (2023). Ultrafast one-pass FASTQ data preprocessing, quality control, and deduplication using fastp. iMeta 2: e107. https://doi.org/10.1002/imt2.107|
| samtools | 1.10.2 |  Danecek, P., Bonfield, J. K., Liddle, J., Marshall, J., Ohan, V., Pollard, M.O., Whitwham, A., Keane, T., McCarthy, S.A., Davies, R.M., Li, H. Twelve years of SAMtools and BCFtools. GigaScience, Volume 10, Issue 2, February 2021, giab008, https://doi.org/10.1093/gigascience/giab008 |
| bcftools | 1.10.2 | Danecek, P., Auton, A., Abecasis, G., Albers, C.A., Banks, E., DePristo, M.A., Handsaker, R., Lunter, G., Marth, G., Sherry, S.T., McVean, G., Durbin, R., and 1000 Genomes Project Analysis Group, (2011). The Variant Call Format and VCFtools. Bioinformatics, 27(15), 2156-2158. https://doi.org/10.1093/bioinformatics/btr330 |
| vcftools | 0.1.17 | Danecek, P., Auton, A., Abecasis, G., Albers, C.A., Banks, E., DePristo, M.A., Handsaker, R., Lunter, G., Marth, G., Sherry, S.T., McVean, G., Durbin, R., and 1000 Genomes Project Analysis Group, (2011). The Variant Call Format and VCFtools. Bioinformatics, 27(15), 2156-2158. https://doi.org/10.1093/bioinformatics/btr330 |
| R | 4.4.1 |  R Core Team (2024). _R: A Language and Environment for Statistical Computing_. R Foundation for Statistical Computing, Vienna, Austria. <https://www.R-project.org/>.|
| tidyverse | 2.0.0 | Wickham H, Averick M, Bryan J, Chang W, McGowan LD, François R, Grolemund G, Hayes A, Henry L, Hester J, Kuhn M, Pedersen TL, Miller E, Bache SM, Müller K, Ooms J, Robinson D, Seidel DP, Spinu V, Takahashi K, Vaughan D, Wilke C, Woo K, Yutani H (2019). “Welcome to the tidyverse.” Journal of Open Source Software, 4(43), 1686. doi:10.21105/joss.01686.|
|  ggplot 2 | 3.5.1 | H. Wickham. ggplot2: Elegant Graphics for Data Analysis. Springer-Verlag New York, 2016. |



